pkg/api/query/v1.go

```diff
@@ -123,6 +126,8 @@ func NewQueryAPI(
 	defaultRangeQueryStep time.Duration,
 	defaultInstantQueryMaxSourceResolution time.Duration,
 	defaultMetadataTimeRange time.Duration,
+	tenantHeader string,
```

Comment:
Why don't we initiate `TenantInfo` where we call `NewQueryAPI`?

Reply:
Because we want to propagate only valuable TenantInfo, we get this when an HTTP request comes and we extract the tenant access according to the configurable tenant Header name (it's not necessary to propagate this label name).